app/models/new_team_event.rb

```diff
@@ -1,14 +1,40 @@
 class NewTeamEvent < AggregatedEvent
   belongs_to_aggregate :team
-  data_attributes :name
+  data_attributes :name, :team_alias
 
-  validate :name_is_present
+  validate :name_is_present, :create_cognito_group
 
   def build_team
     Team.new
   end
 
   def attributes_to_apply
-    { name: name, created_at: created_at }
+    {
+      name: name,
+      team_alias: team_alias,
+      created_at: created_at
+    }
+  end
+
+  def create_cognito_group
+    return if name.blank?
+
+    SelfService.service(:cognito_client).create_group(
+      group_name: team_alias,
+      description: name,
+      user_pool_id: Rails.application.secrets.cognito_user_pool_id
+    )
+  rescue Aws::CognitoIdentityProvider::Errors::InvalidParameterException => e
+    errors.add(:team, "#{I18n.t('team.errors.invalid')} -> #{e.message}")
+  rescue Aws::CognitoIdentityProvider::Errors::ServiceError => e
+    errors.add(:team, "#{I18n.t('team.errors.failed')} -> #{e.message}")
```

Comment:
we probably shouldn't print out the error message to the user, it can be anything from Cognito which could be giving too much details... show generic and log the real one?

Reply:
Good point!